### PR TITLE
JAVA-1607: Add FAQ entry for netty-transport-native-epoll

### PR DESCRIPTION
**Pull request by tolbertam**

### CODE REVIEW

File: faq/README.md [hunk 1]
```diff
@@ -236,6 +236,28 @@ before submitting the next batch.
...
+version of `netty-handler` that the driver depends on.  In such a case, a user may encounter an exception such as the
+one described in [JAVA-1535](https://datastax-oss.atlassian.net/browse/JAVA-1535).
+
+While the epoll transport may in general improve performance, we expect the improvement to be marginal for a lot of use
+cases.  Therefore, if you don't want `netty-transport-native-epoll` to be used by the driver even if the library is
+present in an application's classpath, the most direct way to disable this is to provide the system property value
+`-Dcom.datastax.driver.FORCE_NIO=true` to your application to force the use of the default Netty NIO-based event loop.
+If properly used, the following log message will be logged at INFO on startup:
+
+> Found Netty's native epoll transport in the classpath, but NIO was forced through the FORCE_NIO system property.
```
Comment (newkek) [inline faq/README.md:259]: We can maybe also mention the message logged when NIO is used and epoll is not found: ` Did not find Netty's native epoll transport in the classpath, defaulting to NIO.`

File: faq/README.md [hunk 2]
```diff
@@ -236,6 +236,28 @@ before submitting the next batch.
...
+
+Netty provides [native transport libraries](http://netty.io/wiki/native-transports.html) which generally generate less 
+garbage and improve performance when compared to the default NIO-based transport.
+By default if the driver detects the  `netty-transport-native-epoll` library in its classpath it will attempt to use
+[`EpollEventLoopGroup`](https://netty.io/4.0/api/io/netty/channel/epoll/EpollEventLoopGroup.html) for its underlying
+event loop.
+
+In the usual case this works fine in linux environments.  On the other hand, many users have run into compatibility
+issues when the version of `netty-transport-native-epoll` is not compatible with a version of Netty in an application's
+classpath.  One such case is where an application depends on a version of `netty-all` that is different than the
```
Comment (newkek) [inline faq/README.md:249]: Minor phrasing nit: "One such case is.... In such a case..." - don't even know if it's worth changing, just sounded repetitive but I'm fine like that

File: faq/README.md [hunk 3]
```diff
@@ -236,6 +236,28 @@ before submitting the next batch.
...
+one described in [JAVA-1535](https://datastax-oss.atlassian.net/browse/JAVA-1535).
+
+While the epoll transport may in general improve performance, we expect the improvement to be marginal for a lot of use
+cases.  Therefore, if you don't want `netty-transport-native-epoll` to be used by the driver even if the library is
+present in an application's classpath, the most direct way to disable this is to provide the system property value
+`-Dcom.datastax.driver.FORCE_NIO=true` to your application to force the use of the default Netty NIO-based event loop.
+If properly used, the following log message will be logged at INFO on startup:
+
+> Found Netty's native epoll transport in the classpath, but NIO was forced through the FORCE_NIO system property.
+
```
Comment (newkek) [inline faq/README.md:260]: Maybe worth mentioning that there is a programmatic way to find out if the driver is using Epoll or not, via `NettyUtil.isEpollAvailable()`

**Review by newkek — APPROVED** [review]
sgtm! Had some minor comments.

File: faq/README.md (hunk 3 above)
Comment (tolbertam) [reply faq/README.md:260]: good call, will do that.

**Comment by tolbertam** [comment]
Thanks for your feedback, will update shortly!

**Pull request merged**